### PR TITLE
fix: Making sure all Sentry processing runs in the collaborative background thread pool.

### PR DESCRIPTION
**Pull request by adrien-coye**
### Notes
- Reduce direct calls to `SentrySDK` to bare minimum
- `captureError` and `addBreadcrumb` are done thanks to the `SentryDebug` type.
- `SentryDebug` takes care of dispatching `SentrySDK` calls into the collaborative thread pool.
- Reduced the number of files that import `SentrySDK` directly.

### CODE REVIEW

**Review by PhilippeWeidmann — CHANGES_REQUESTED** [review]
Only applying Task doesn't switch threads. We will discuss it live.

**Comment by adrien-coye** [comment]
@PhilippeWeidmann Using `Task` does … depending on the actor context. Here it does work as I'm outside the main actor as we did discuss yesteday.

I did use the `Task.detached` to leave the main actor, when needed, but I will refactor aroud it to not use it anymore for my code to be cleaner. I'll prepare a proper set of slide for next monday to sort this out for the team for good 👍

File: kDrive/UI/Controller/Menu/ParameterTableViewController.swift [hunk 1]
```diff
@@ -217,7 +217,9 @@ extension ParameterTableViewController: DeleteAccountDelegate {
     }
 
     func didFailDeleteAccount(error: InfomaniakLoginError) {
-        SentrySDK.capture(error: error)
+        Task.detached {
```
Comment (adrien-coye) [inline kDrive/UI/Controller/Menu/ParameterTableViewController.swift:220]: TODO: refactor

File: kDriveCore/Data/Api/DriveApiFetcher.swift [hunk 2]
```diff
@@ -476,14 +476,18 @@ class SyncedAuthenticator: OAuthAuthenticator {
     ) {
         // Only resolve locally to break init loop
         accountManager.refreshTokenLockedQueue.async {
-            SentrySDK
-                .addBreadcrumb((credential as ApiToken)
-                    .generateBreadcrumb(level: .info, message: "Refreshing token - Starting"))
-
-            if !KeychainHelper.isKeychainAccessible {
+            Task {
```
Comment (adrien-coye) [inline kDriveCore/Data/Api/DriveApiFetcher.swift:479]: TODO use SentryDebug

File: kDriveCore/Data/Api/DriveApiFetcher.swift [hunk 3]
```diff
@@ -526,25 +524,25 @@ class SyncedAuthenticator: OAuthAuthenticator {
             self.tokenable.refreshToken(token: credential) { token, error in
                 // New token has been fetched correctly
                 if let token {
-                    SentrySDK
-                        .addBreadcrumb(token
-                            .generateBreadcrumb(level: .info, message: "Refreshing token - Success with remote"))
+                    let message = "Refreshing token - Success with remote"
+                    let breadcrumbMetadata = (credential as ApiToken).breadcrumbMetadata()
```
Comment (adrien-coye) [inline kDriveCore/Data/Api/DriveApiFetcher.swift:528]: TODO: remove

File: kDriveCore/Utils/KeychainHelper.swift [hunk 4]
```diff
@@ -124,7 +123,14 @@ public enum KeychainHelper {
                     ]
                     resultCode = SecItemUpdate(queryUpdate as CFDictionary, attributes as CFDictionary)
                     DDLogInfo("Successfully updated token ? \(resultCode == noErr)")
-                    SentrySDK.addBreadcrumb(token.generateBreadcrumb(level: .info, message: "Successfully updated token"))
+
+                    let metadata = token.breadcrumbMetadata()
```
Comment (adrien-coye) [inline kDriveCore/Utils/KeychainHelper.swift:127]: missing `resultCode`

**Pull request merged**